src/projects/EnsureThat/Enforcers/AnyArg.cs

```diff
@@ -1,10 +1,19 @@
+using System;
 using EnsureThat.Annotations;
 using JetBrains.Annotations;
 
 namespace EnsureThat.Enforcers
 {
     public sealed class AnyArg
     {
+        [NotNull]
+        [ContractAnnotation("value:null => halt")]
+        public T HasValue<T>([NoEnumeration, ValidatedNotNull] T value, [InvokerParameterName] string paramName = null, OptsFn optsFn = null) => value switch
+        {
+            ValueType _ => value,
```

Comment:
Looks like `ValueType` != `struct` - this syntax is new to me so I can only cursorily look over it, since I would've expected int? (I think Nullable<T> is a struct?) to be ValueType, but your tests show that must not be true